src/expr/src/scalar/mod.rs

```diff
@@ -645,6 +645,22 @@ impl MirScalarExpr {
         matches!(self, MirScalarExpr::Literal(Err(_), _typ))
     }
 
+    pub fn contains_error_if_null(&self) -> Result<bool, RecursionLimitError> {
+        let mut result = false;
+        self.visit_post(&mut |expr: &MirScalarExpr| {
```

Comment:
nit: I suggest using `MirRelationExpr::visit_pre` (which is not fallable) here. You can also roll your own worklist iteration which will allow you to exit early and not traverse the entire tree once you see an `ErrorIfNull` call.

Reply:
But this is a `MirScalarExpr`, which doesn't seem to have an infallible `visit_pre`.

Also, I tried writing a worklist iteration, but got stuck:
```
    pub fn contains_error_if_null(&self) -> bool {
        let mut worklist = vec![self];
        while let Some(expr) = worklist.pop() {
            if matches!(
                expr,
                MirScalarExpr::CallVariadic {
                    func: VariadicFunc::ErrorIfNull,
                    ..
                }
            ) {
                return true;
            }
            expr.visit_children(|child| worklist.push(child));
        }
        false
    }
```
This gives me
```
error[E0521]: borrowed data escapes outside of closure
   --> src/expr/src/scalar/mod.rs:660:41
    |
649 |         let mut worklist = vec![self];
    |             ------------ `worklist` declared here, outside of the closure body
...
660 |             expr.visit_children(|child| worklist.push(child));
    |                                  -----  ^^^^^^^^^^^^^^^^^^^^ `child` escapes the closure body here
    |                                  |
    |                                  `child` is a reference that is only valid in the closure body
```
which I guess makes sense, since the compiler can't know that `visit_children` is passing references to stuff that is inside `expr` and not to stuff that it is making up itself.

Note that worklist iterations on `MirRelationExpr` use the `children()` method instead of `visit_children`, but we don't seem to have `children()` on `MirScalarExpr`. Should I maybe write it?

Reply:
Ok, I implemented `MirScalarExpr::children()`. Could you please check?